src/exports/arcdps.rs

```diff
@@ -0,0 +1,497 @@
+    static MUMBLE_IDENTITY: RefCell<MumbleIdentity> = RefCell::new(MumbleIdentity::new());
+}
+
+fn update_mumble_link() {
+    let ml = match mumble_ptr() {
+        Some(ml) => ml,
+        None => return,
+    };
+
+    let update = MUMBLE_IDENTITY.with_borrow_mut(|identity| {
+        match identity.update(&ml) {
+            true => Some((*identity.identity).clone()),
+            false => None,
+        }
+    });
+
+    if let Some(update) = update {
+        receive_mumble_identity(update);
+    }
+}
+
+#[cfg(feature = "extension-arcdps-codegen")]
+pub(crate) fn cb_imgui(ui: &imgui::Ui, ingame: bool) {
+    IS_INGAME.store(ingame, Ordering::Relaxed);
+
+    update_mumble_link();
+
+    #[cfg(feature = "space")] {
+        crate::render_space(ui);
+    }
+
+    render_overlay(ui);
+}
+
+pub(crate) fn cb_options_end(ui: &imgui::Ui) {
+    ui.text("WORK IN PROGRESS");
+
+    ui.checkbox("Check for updates", &mut false);
+
+    ui.new_line();
+    let all_windows = [
+        WINDOW_PRIMARY,
+        WINDOW_TIMERS,
+        #[cfg(feature = "markers")]
+        crate::WINDOW_MARKERS,
+    ];
+    for window in all_windows {
+        let _id = ui.push_id(window);
+        let singular = window.strip_suffix("s");
+        let name = crate::LANGUAGE_LOADER.get(&format!("{}-window-toggle", singular.unwrap_or(window)));
+        if ui.button(name) {
+            control_window(window, None);
+        }
+        ui.same_line();
+        ui.text("Keybind: ");
+        ui.same_line();
+        ui.text_disabled("ALT+SHIFT+");
+        ui.same_line();
+        if ui.button("BIND") {
+            log::warn!("TODO: keybind settings");
+        }
+        if window == WINDOW_PRIMARY {
+            let desc = crate::LANGUAGE_LOADER.get(&format!("{window}-window-toggle-text"));
+            ui.text_disabled(desc);
+        }
+        ui.separator();
+    }
+
+    let selected_language = game_language()
+        .map(game_language_id)
+        .unwrap_or("");
+    if let Some(languages) = ui.begin_combo("Language", selected_language) {
+        let mut new_language = None;
+        for l in crate::LANGUAGES_GAME {
+            let id = game_language_id(l);
+            let selected = imgui::Selectable::new(id)
+                .selected(selected_language == id)
+                .build(ui);
+            if selected {
+                new_language = Some(Ok(l));
+            }
+        }
+        for id in crate::LANGUAGES_EXTRA {
+            let selected = imgui::Selectable::new(id)
+                .selected(selected_language == id)
+                .build(ui);
+            if selected {
+                new_language = Some(Err(id));
+            }
+        }
+        languages.end();
+
+        if let Some(new_language) = new_language {
+            log::warn!("TODO: language selection");
+        }
+    }
+}
+
+#[cfg(feature = "extension-arcdps-codegen")]
+pub(crate) fn cb_wnd_filter(keycode: usize, key_down: bool, prev_key_down: bool) -> bool {
+    true
```

Comment:
this is where hotkey triggers go btw
(but they also need a place in settings to load from and save to? since nexus was used previously to manage it for you afaict)